# 1. Marketing Campaign Data Cleaning

## Introduction

This project focuses on cleaning a dataset that tracks the performance of various marketing campaigns. The dataset includes detailed metrics such as impressions, clicks, leads, orders, and revenue generated across different campaign categories. The goal is to prepare the data for further analysis, enabling actionable insights into campaign performance and effectiveness.

### Objectives

1. **Data Cleaning**:
   - Ensure data consistency by checking for and removing duplicates.
   - Convert date columns to a usable datetime format.
   - Validate the absence of missing values in critical fields.

2. **Data Preparation**:
   - Standardize the dataset to ensure readiness for exploratory data analysis (EDA) and advanced modeling.
   - Save the cleaned dataset for future use.

---

In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv(r"C:\Users\Zana\Desktop\portfolio_projects\project_3\Marketing.csv")
print(df.head())

   id      c_date    campaign_name    category  campaign_id  impressions  \
0   1  2021-02-01   facebook_tier1      social       349043       148263   
1   2  2021-02-01   facebOOK_tier2      social       348934       220688   
2   3  2021-02-01       google_hot      search     89459845        22850   
3   4  2021-02-01      google_wide      search       127823       147038   
4   5  2021-02-01  youtube_blogger  influencer        10934       225800   

   mark_spent  clicks  leads  orders  revenue  
0     7307.37    1210     13       1   4981.0  
1    16300.20    1640     48       3  14962.0  
2     5221.60     457      9       1   7981.0  
3     6037.00    1196     24       1   2114.0  
4    29962.20    2258     49      10  84490.0  


In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308 entries, 0 to 307
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             308 non-null    int64  
 1   c_date         308 non-null    object 
 2   campaign_name  308 non-null    object 
 3   category       308 non-null    object 
 4   campaign_id    308 non-null    int64  
 5   impressions    308 non-null    int64  
 6   mark_spent     308 non-null    float64
 7   clicks         308 non-null    int64  
 8   leads          308 non-null    int64  
 9   orders         308 non-null    int64  
 10  revenue        308 non-null    float64
dtypes: float64(2), int64(6), object(3)
memory usage: 26.6+ KB


In [4]:
df.isnull()

id  c_date  campaign_name  category  campaign_id  impressions  \
0    False   False          False     False        False        False   
1    False   False          False     False        False        False   
2    False   False          False     False        False        False   
3    False   False          False     False        False        False   
4    False   False          False     False        False        False   
..     ...     ...            ...       ...          ...          ...   
303  False   False          False     False        False        False   
304  False   False          False     False        False        False   
305  False   False          False     False        False        False   
306  False   False          False     False        False        False   
307  False   False          False     False        False        False   

     mark_spent  clicks  leads  orders  revenue  
0         False   False  False   False    False  
1         False   False  False   False    False  
2         False   False  False   False    False  
3         False   False  False   False    False  
4         False   False  False   False    False  
..          ...     ...    ...     ...      ...  
303       False   False  False   False    False  
304       False   False  False   False    False  
305       False   False  False   False    False  
306       False   False  False   False    False  
307       False   False  False   False    False  

[308 rows x 11 columns]

In [6]:
# Check for missing values
df.isnull().sum()

id               0
c_date           0
campaign_name    0
category         0
campaign_id      0
impressions      0
mark_spent       0
clicks           0
leads            0
orders           0
revenue          0
dtype: int64

In [7]:
df.drop_duplicates()

id      c_date         campaign_name    category  campaign_id  \
0      1  2021-02-01        facebook_tier1      social       349043   
1      2  2021-02-01        facebOOK_tier2      social       348934   
2      3  2021-02-01            google_hot      search     89459845   
3      4  2021-02-01           google_wide      search       127823   
4      5  2021-02-01       youtube_blogger  influencer        10934   
..   ...         ...                   ...         ...          ...   
303  304  2021-02-28       instagram_tier2      social       983498   
304  305  2021-02-28  facebook_retargeting      social      4387490   
305  306  2021-02-28          facebook_lal      social       544756   
306  307  2021-02-28     instagram_blogger  influencer       374754   
307  308  2021-02-28        banner_partner       media        39889   

     impressions  mark_spent  clicks  leads  orders  revenue  
0         148263     7307.37    1210     13       1   4981.0  
1         220688    16300.20    1640     48       3  14962.0  
2          22850     5221.60     457      9       1   7981.0  
3         147038     6037.00    1196     24       1   2114.0  
4         225800    29962.20    2258     49      10  84490.0  
..           ...         ...     ...    ...     ...      ...  
303       775780      760.75    1024      4       0      0.0  
304         1933      224.81      58      0       0      0.0  
305        25840     6844.80     248      5       1   1491.0  
306        94058     4845.65     594     12       1   5008.0  
307      8490000     6822.62     849     18       2   7030.0  

[308 rows x 11 columns]

In [8]:
# Convert the 'c_date' column to datetime format
df['c_date'] = pd.to_datetime(df['c_date'])

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308 entries, 0 to 307
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   id             308 non-null    int64         
 1   c_date         308 non-null    datetime64[ns]
 2   campaign_name  308 non-null    object        
 3   category       308 non-null    object        
 4   campaign_id    308 non-null    int64         
 5   impressions    308 non-null    int64         
 6   mark_spent     308 non-null    float64       
 7   clicks         308 non-null    int64         
 8   leads          308 non-null    int64         
 9   orders         308 non-null    int64         
 10  revenue        308 non-null    float64       
dtypes: datetime64[ns](1), float64(2), int64(6), object(2)
memory usage: 26.6+ KB


In [11]:
# Save the cleaned DataFrame to a CSV file
df.to_csv('cleaned_marketing.csv', index=False)

## Key Outcomes

1. **Data Inspection**:
   - The dataset contains **308 rows** and **11 columns**.
   - There are **no missing values** across any columns, ensuring a complete dataset for analysis.
   - No duplicate rows were found, indicating that the dataset is free of redundancy.

2. **Data Cleaning**:
   - The `c_date` column was successfully converted to datetime format for accurate temporal analysis.
   - Column data types were validated:
     - Numerical data types: `int64` and `float64` for metrics like impressions, clicks, leads, and revenue.
     - Categorical data types: `object` for campaign names and categories.
     - Datetime: Properly converted for the `c_date` column.

3. **Prepared Dataset**:
   - The cleaned dataset is saved as **`cleaned_marketing.csv`** for future use in:
     - Exploratory analysis of campaign performance.
     - Revenue attribution and optimization modeling.
     - Insights into campaign effectiveness by category and time period.

---